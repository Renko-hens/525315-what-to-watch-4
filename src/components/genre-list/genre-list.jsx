import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import GenreItem from "../genre-item/genre-item.jsx";
import {ActionCreator} from "../../reducer.js";
import {Genres} from "../../const/const.js";

const GenresArray = Object.values(Genres);

const GenreList = (props) => {
  const {genre, onGenreLinkClick} = props;
  return (
    <ul className="catalog__genres-list">
      {GenresArray.map((genreItem, i) => {
        return (
          <GenreItem
            key={`genre-${i}`}
            genre={genreItem}
            activeGenre={genre}
            onGenreLinkClick={onGenreLinkClick}
          />);
      })}
    </ul>
  );
};

GenreList.propTypes = {
  genre: PropTypes.string.isRequired,
  onGenreLinkClick: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  genre: state.genre,
});

const mapDispatchToProps = (dispatch) => ({
  onGenreLinkClick(genre) {
    dispatch(ActionCreator.changeGenre(genre));
    dispatch(ActionCreator.getMoviesByGenre(genre));
  },
});

export {GenreList};
export default connect(mapStateToProps, mapDispatchToProps)(GenreList);
