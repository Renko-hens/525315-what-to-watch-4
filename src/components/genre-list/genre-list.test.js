import React from "react";
import renderer from "react-test-renderer";
import {GenreList} from "./genre-list.jsx";

describe(`Renders Genre List component`, () => {
  it(`Render component Genre List`, () => {
    const tree = renderer.create(
        <GenreList
          genre={`All genres`}
          onGenreLinkClick={() => {}}
        />
    ).toJSON;

    expect(tree).toMatchSnapshot();
  });
});
