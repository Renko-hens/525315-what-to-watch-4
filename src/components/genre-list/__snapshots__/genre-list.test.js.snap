// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Genre List component Render component Genre List 1`] = `[Function]`;
